src/client.ts

```diff
@@ -43,39 +32,93 @@ export type AcmeGroup = Opaque<any, 'AcmeGroup'>;
-    // authentication works with the provided credentials, throw an err if
-    // authentication fails
-    const request = new Promise<void>((resolve, reject) => {
-      http.get(
-        {
-          hostname: 'localhost',
-          port: 443,
-          path: '/api/v1/some/endpoint?limit=1',
-          agent: false,
-          timeout: 10,
-        },
-        (res) => {
-          if (res.statusCode !== 200) {
-            reject(new Error('Provider authentication failed'));
-          } else {
-            resolve();
-          }
-        },
-      );
+  private withBaseUri(path: string): string {
+    return `https://${this.config.cybereasonHost}:${this.config.cybereasonPort}/${path}`;
+  }
+
+  private async getRequest(uri: string, cookie: string): Promise<Response> {
+    const response = await fetch(uri, {
+      headers: {
+        cookie,
+      },
     });
+    if (!response.ok) {
+      throw new IntegrationProviderAPIError({
+        endpoint: uri,
+        status: response.status,
+        statusText: response.statusText,
+      });
+    }
+    return response;
+  }
 
-    try {
-      await request;
-    } catch (err) {
-      throw new IntegrationProviderAuthenticationError({
-        cause: err,
-        endpoint: 'https://localhost/api/v1/some/endpoint?limit=1',
-        status: err.status,
-        statusText: err.statusText,
+  private async postRequest(
+    uri: string,
+    body: any,
+    dontRedirect?: boolean,
+  ): Promise<Response> {
+    const redirect = dontRedirect ? 'manual' : 'follow';
+
+    const response = await fetch(uri, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: querystring.stringify(body),
+      redirect,
+    });
+    if (!(response.ok || (response.status === 302 && dontRedirect))) {
+      throw new IntegrationProviderAPIError({
+        endpoint: uri,
+        status: response.status,
+        statusText: response.statusText,
       });
     }
+    return response;
+  }
+
+  public async login(): Promise<string> {
+    const loginRoute = this.withBaseUri('login.html');
+    const res = await this.postRequest(
+      loginRoute,
+      {
+        username: this.config.cybereasonId,
+        password: this.config.cybereasonPassword,
+      },
+      true,
+    );
+    return (
+      res.headers
+        .get('set-cookie')
+        ?.split('; ')
+        .filter((item) => item.search('JSESSIONID') >= 0)[0] || ''
+    );
+  }
+
+  public async logout(cookie: string): Promise<void> {
+    const logoutRoute = this.withBaseUri('logout');
+    await this.getRequest(logoutRoute, cookie);
+  }
+
+  public async verifyAuthentication(): Promise<void> {
+    const cookie = await this.login();
+
+    // NOTE: This only works with 'System admin' role
+    // no API endpoint is available for all roles, need to find
+    // a better endpoint to test the cookie that is not hindered
+    // by role permissions.
```

Comment:
Permissions table: https://nest.cybereason.com/documentation/api-documentation/all-versions/required-permissions

A false authentication would occur when the user has no permission to the testing endpoint. We need an endpoint that is available for all roles but no such thing exists.

Reply:
Yep, I see, thanks for pointing this out. Let's revisit this after you've had the chance to interact with more endpoints.